core/network.py

```diff
@@ -72,111 +139,72 @@ def new_network_events(configuration):
-                            netaddr.valid_ipv6(ip_dest)) \
-                                and ip_dest not in ignore_ip_addresses \
-                                and ip_src not in ignore_ip_addresses \
-                                and port_dest not in ignore_ports \
-                                and port_src not in ignore_ports:
-                            # ignored ip addresses and ports in python -fix later
-                            # check if the port is in selected module
-
-                            if (port_dest in honeypot_ports or
-                                    port_src in honeypot_ports):
-                                if port_dest in honeypot_ports:
-                                    insert_to_honeypot_events_queue(
-                                        HoneypotEvent(
-                                            ip_dest=ip_dest,
-                                            port_dest=port_dest,
-                                            ip_src=ip_src,
-                                            port_src=port_src,
-                                            module_name=selected_module,
-                                            machine_name=machine_name
-                                        )
-                                    )
-                            else:
-                                insert_to_network_events_queue(
-                                    NetworkEvent(
-                                        ip_dest=ip_dest,
-                                        port_dest=port_dest,
-                                        ip_src=ip_src,
-                                        port_src=port_src,
-                                        machine_name=machine_name
-                                    )
-                                )
-                    except Exception:
-                        pass
-                    # check if event shows an IP
-            time.sleep(0.001)
-            # todo: is sleep(0.001) fastest/best?
-            # it means it could get 1000 packets per second(maximum) from
-            # tshark
-            # how could we prevent the DDoS attacks in here
-            # and avoid submitting in MongoDB? should we?
-    except Exception as _:
-        del _
+
+        if store_captured_traffic:
+            capture = pyshark.LiveCapture(
+                interface='any',
+                display_filter=display_filter,
+                output_file=output_file_path
+            )
+        else:
+            capture = pyshark.LiveCapture(
+                interface='any',
+                display_filter=display_filter
+            )
+        # Debug option for pyshark capture
+        # capture.set_debug()
+
+        # Applied on every packet captured by pyshark LiveCapture
+        capture.apply_on_packets(process_packet)
+
+    except Exception as _e:
+        # _e is Runtime, we've to convert it to str before calling error()
+        error(str(_e))
```

Comment:
for some reason i am still getting an error here

```
batman@Batmans-MacBook-Pro:~/D/G/OWASP-Honeypot:pr/103$ sudo python3 ohp.py -m ssh/weak_password

      ______          __      _____ _____
     / __ \ \        / /\    / ____|  __ \
    | |  | \ \  /\  / /  \  | (___ | |__) |
    | |  | |\ \/  \/ / /\ \  \___ \|  ___/
    | |__| | \  /\  / ____ \ ____) | |
     \____/   \/  \/_/    \_\_____/|_|
                      _    _                        _____      _
                     | |  | |                      |  __ \    | |
                     | |__| | ___  _ __   ___ _   _| |__) |__ | |_
                     |  __  |/ _ \| "_ \ / _ \ | | |  ___/ _ \| __|
                     | |  | | (_) | | | |  __/ |_| | |  | (_) | |_
                     |_|  |_|\___/|_| |_|\___|\__, |_|   \___/ \__|
                                               __/ |
                                              |___/

[+] [2020-07-24 13:10:36] OWASP Honeypot started ...
[+] [2020-07-24 13:10:36] loading modules ssh/weak_password
[+] [2020-07-24 13:10:36] checking for conflicts in ports
[+] [2020-07-24 13:10:36] port 22 selected for ssh/weak_password
[+] [2020-07-24 13:10:36] creating image ohp_sshserver_weak_password
[+] [2020-07-24 13:10:51] image ohp_sshserver_weak_password created
[+] [2020-07-24 13:10:51] container ohp_sshserver_weak_password started, forwarding 0.0.0.0:22 to 172.25.0.2:22
[+] [2020-07-24 13:10:51] new_network_events thread started
[+] [2020-07-24 13:10:51] all selected modules started: ssh/weak_password
[X] [2020-07-24 13:10:51] Cannot add child handler, the child watcher does not have a loop attached
[X] [2020-07-24 13:10:52] Interrupting the application because network capturing thread is not alive!
[+] [2020-07-24 13:10:52] killing new_network_events_thread
[+] [2020-07-24 13:10:52] killing insert_events_in_bulk_thread
[+] [2020-07-24 13:10:52] killing container ohp_sshserver_weak_password
[+] [2020-07-24 13:10:53] removing container ohp_sshserver_weak_password
[+] [2020-07-24 13:10:53] removing image ohp_sshserver_weak_password
[+] [2020-07-24 13:10:53] finished.
```

@ChakshuGupta whats your OS and python version?

Reply:
ubuntu 20.04
Python 3.8